sdks/python/apache_beam/tools/utils.py

```diff
@@ -32,3 +40,79 @@ def check_compiled(module):
         "Profiling uncompiled code.\n"
         "To compile beam, run "
         "'pip install Cython; python setup.py build_ext --inplace'")
+
+
+def run_benchmarks(benchmark_suite, verbose=True):
+  """Runs benchmarks, and collects execution times.
+
+  A simple instrumentation to run a callable several times, collect and print
+  its execution times.
+
+  Args:
+    benchmark_suite: A list of named tuples that describe benchmarks.
+                     Each tuple should have following key-value pairs:
+        benchmark: a callable that takes an int argument - benchmark size,
+          and returns a callable. A returned callable must run the code being
+          benchmarked on an input of specified size.
+
+          For example, one can implement a benchmark as:
+
+          class MyBenchmark(object):
+            def __init__(self, size):
+              [do necessary initialization]
+
+            def __call__(self):
+              [run the code in question]
+
+        size: int, a size of the input. Aggregated per-element metrics
+           are counted based on the size of the input.
+    num_runs: int, number of times to run each benchmark.
+    verbose: bool, whether to print benchmark results to stdout.
+
+  Returns:
+    A dictionary of the form string -> list of floats. Keys of the dictionary
+    are benchmark names, values are execution times in seconds for each run.
+  """
+
+  def get_name(benchmark_config):
+    return getattr(benchmark_config.benchmark, '__name__',
+                   str(benchmark_config.benchmark))
+
+  def run(benchmark_fn, size):
+    # Contain each run of a benchmark inside a function so that any temporary
+    # objects can be garbage-collected after the run.
+    benchmark_instance_callable = benchmark_fn(size)
+    start = time.time()
+    _ = benchmark_instance_callable()
+    return time.time() - start
+
+  cost_series = collections.defaultdict(list)
+  for benchmark_config in benchmark_suite:
+    name = get_name(benchmark_config)
+    num_runs = benchmark_config.num_runs
+    size = benchmark_config.size
```

Comment:
Or `benchmark, size, num_runs = benchmark_config` to allow "ordinary" tuples as well.